client/js/widgets/dice.js

```diff
@@ -0,0 +1,384 @@
+class Dice extends Widget {
+  constructor(id) {
+    super(id);
+
+    this.facesElement = document.createElement('div');
+    this.facesElement.className = 'diceFaces';
+
+    this.domElement.appendChild(this.facesElement);
+
+    this.addDefaults({
+      width: 50,
+      height: 50,
+      typeClasses: 'widget dice',
+      classes: 'shaking',
+      clickable: true,
+      movable: false,
+      layer: 1,
+
+      borderRadius: '16%',
+      pipColor: 'black',
+      backgroundColor: 'white',
+      borderColor: null,
+
+      options: [ 1, 2, 3, 4, 5, 6 ],
+      activeFace: 0,
+      rollCount: 0,
+
+      pipSymbols: null,
+      shape3d: false
+    });
+  }
+
+  activeFace() {
+    return mod(Math.round(this.get('activeFace')), this.faceElements.length) || 0;
+  }
+
+  animate() {
+    if(!this.facesElement || !this.faceElements.length)
```

Comment:
Pretty sure this should be `!this.faceElements`

Reply:
No, I wanted to check if there are more than 0 faces because the code can't handle that, I think.